src/gen9_vdenc.c

```diff
@@ -2810,11 +2810,11 @@ gen95_vdenc_vdenc_walker_state(VADriverContextP ctx,
     int luma_log2_weight_denom, weighted_pred_idc;
 
     slice_hor_pos = slice_param->macroblock_address % vdenc_context->frame_width_in_mbs;
-    slice_ver_pos = slice_param->macroblock_address / vdenc_context->frame_height_in_mbs;
+    slice_ver_pos = slice_param->macroblock_address / vdenc_context->frame_width_in_mbs;
```

Comment:
The AVC_SLICE_STATE GPU commands need the two kind of coord for the slice. 
       >slice_x/slice_y for the current slice
       >slice_x/slice_y for the next slice

So we need to calculate them based on macroblock_address. 
The current calculation is wrong especially for multi-slices. Of course the issue is hidden for the single slice that the macroblock_address is zero.